frontend/public/components/factory/list-page.jsx

```diff
@@ -25,66 +25,43 @@ import {
 } from '../utils';
 import { FilterToolbar } from '../filter-toolbar';
 
-/** @type {React.SFC<{disabled?: boolean, label?: string, onChange: React.ChangeEventHandler<any>, defaultValue?: string, value?: string, placeholder?: string, autoFocus?: boolean,}}>} */
-export const TextFilter = ({
-  label,
-  onChange,
-  defaultValue,
-  style,
-  className,
-  value,
-  placeholder = `Filter ${label}...`,
-  autoFocus = false,
-}) => {
-  const input = React.useRef();
-  const onKeyDown = (e) => {
-    const { nodeName } = e.target;
-    if (
-      nodeName === 'INPUT' ||
-      nodeName === 'TEXTAREA' ||
-      e.key !== KEYBOARD_SHORTCUTS.focusFilterInput
-    ) {
-      return;
-    }
-
-    e.stopPropagation();
-    e.preventDefault();
-    input.current.focus();
-  };
-
-  React.useEffect(() => {
-    window.addEventListener('keydown', onKeyDown);
-    // Remove event listeners on cleanup
-    return () => {
-      window.removeEventListener('keydown', onKeyDown);
-    };
-  }, []);
+/** @type {React.SFC<{disabled?: boolean, label?: string, onChange: (value: string) => void;, defaultValue?: string, value?: string, placeholder?: string, autoFocus?: boolean, onFocus?:any, name?:string, id?: string, onKeyDown?: any, parentClassName?: string }}>} */
+export const TextFilter = (props) => {
+  const {
+    label,
+    defaultValue,
+    className,
+    value,
+    placeholder = `Filter ${label}...`,
+    autoFocus = false,
+    onChange,
+    parentClassName,
+  } = props;
+  const { ref } = useDocumentListener();
 
   return (
-    <div className="has-feedback">
-      <input
-        ref={input}
-        autoCapitalize="none"
-        className={classNames('pf-c-form-control co-text-filter', className)}
+    <div className={classNames('has-feedback', parentClassName)}>
+      <TextInput
+        {...props}
+        className={classNames('co-text-filter', className)}
         data-test-id="item-filter"
-        value={value}
-        defaultValue={defaultValue}
-        onChange={onChange}
-        onKeyDown={(e) => e.key === 'Escape' && e.target.blur()}
+        aria-label={placeholder}
         placeholder={placeholder}
-        style={style}
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars, no-unused-vars
+        onChange={(val, _event) => onChange(val)}
```

Comment:
Sorry, I wasn't clear. I was suggesting

```suggestion
        onChange={onChange}
```

and changing the callback signature to take an event to match the PF `onChange` callback. (It can be ignored by callers who don't care.)

Reply:
Thanks for the clarification. Done!